apis/v1beta1/object_reference_types.go

```diff
@@ -145,3 +145,42 @@ type BackendObjectReference struct {
 	// +optional
 	Port *PortNumber `json:"port,omitempty"`
 }
+
+// ObjectReference identifies an API object including its namespace, with no
+// default kind.
+//
+// The API object must be valid in the cluster; the Group and Kind must
+// be registered in the cluster for this reference to be valid.
+//
+// References to objects with invalid Group and Kind are not valid, and must
+// be rejected by the implementation, with appropriate Conditions set
+// on the containing object.
+type ObjectReference struct {
+	// Group is the group of the referent.  For example, "gateway.networking.k8s.io".
+	// When unspecified or empty string, core API group is inferred.
+	//
+	// +kubebuilder:validation:Required
+	Group *Group `json:"group"`
+
+	// Kind is the kind of the referent.  For example, "ConfigMap".
+	//
+	// +kubebuilder:validation:Required
+	Kind *Kind `json:"kind"`
```

Comment:
According to [API conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required), this field should not be a pointer if it's required.